### ISSUE TITLE
File browser: Make Folder and file icons visually better distinct (e.g. using color)

### ISSUE THREAD

**Issue by camoz** [gsantner/markor#2186]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **enhancement/feature request**. Not a [bug or question](https://github.com/gsantner/markor/issues/new/choose).
- [X] The topic is **not** already reported at [Issues](https://github.com/gsantner/markor/issues?q=#js-issues-search). _(I've searched it)_.
- [X] Markor **is** up to date. See [Releases](https://github.com/gsantner/markor/tags) for the latest version. Updates are available from [F-Droid](https://f-droid.org/en/packages/net.gsantner.markor/) and GitHub.
- [X] The wanted feature/enhancement is not present in the latest development version (git master). (Please [download](https://nightly.link/gsantner/markor/workflows/build-android-project/master) and try the test version of Markor, named **Marder**. Don't worry; Markor and Marder appear as completely separate applications. You can install both side-by-side, and Markor's settings are not touched. If your desired feature is present, you don't need to open this issue. The change will be part of the next Markor update.)

### Description

IMO it would be helpful if the folder/file icons were visually more distinct.

When scrolling a folder with many folders and files, it's easy to roughly navigate to the part where folders end and files begin (assuming folders are sorted before files). But since the folder and file icons are so similar (same color grey, very similar shape), the eyes have to scan from top to bottom to find the first file.

An option would be to be able to set the color for folder and file icons.

### Information

Android version: 14 (GrapheneOS)
Device: Pixel 8
App Version: 2.11.0

### Source

F-Droid

### Format / File type

Not specific

**Comment by gsantner** [comment]
> An option would be to be able to set the color for folder and file icons.

That breaks as soon "folder first" disabled. Not a good idea.

Overall my idea is that the main focus is the filename. The rest is meta info and hence should be less distracting.

**Comment by camoz** [comment]
> That breaks as soon "folder first" disabled. Not a good idea.

How does it break with "folder first" disabled? I'd argue this is even the more important use case: Locating a single folder (alternatively a file) in a directory with many files (alternatively folders). If folder/file icons were more easily distinguishable, the cognitive load for this becomes almost zero. Right now it's unnecessarily cumbersome.

**Comment by gsantner** [comment]
> When scrolling a folder with many folders and files, it's easy to roughly navigate to the part where folders end and files begin 

no files/folder specific sorting, "first folder/file" gets meaningless as the file/folder info is not taken into account.

**Comment by camoz** [comment]
Not sure I understand what you mean. What I mean by my previous comment is that with icon colors, it's easier to spot a file between many folders, or a folder between many files (in case the user has "folder first" disabled, like you said in https://github.com/gsantner/markor/issues/2186#issuecomment-1870191200).

IMO in both cases it helps readability, at no (or negligible) cost. The same like it helps to have the output of `ls` colored, to be able to spot folders easily. FWIW, the vast majority of modern file browser makes folders easily distinguishable from files, and in most cases use color for that.

**Comment by gsantner** [comment]
@camoz 
At the core, you are asking to customizable icon colors at this place.

We have a quite outdated color picker in use here that is not developed anymore and we happen to have it around still. So as first step for further color customization it would be required to replace that with something newer. Then it can be thought of implementing additional color customization.

Just take my view on this: Customizable colors can and do break the app. User choses wrong color, can't see anything anymore. Color swaps automatically from light to dark mode? Boom, broken. Requires two sets of colors. There are so many possibilities. Hence I prefer to keep it simple, have a sane default, whats here for years now. And not everybody can't be happy about that, but think the majority is OK with it.

So TL;DR: If you want to make that customizable, feel free to look up alternatives for color picking, implement that as replacement in Markor and add picking for said colors. Though, make sure not to break the app. These colors are used on multiple places. We don't implement colors for every single part of the app.

**Comment by camoz** [comment]
> We have a quite outdated color picker in use here that is not developed anymore and we happen to have it around still. So as first step for further color customization it would be required to replace that with something newer. Then it can be thought of implementing additional color customization.

Hm I did not know that the color picker is outdated. Personally, I thought of "just" adding four entries to `Settings -> Other -> Basic color scheme` (both file+folder color for light/dark theme) for now. If you say there needs to be a new color picker first, it's your decision of course! I would just appreciate it if this issue will be kept open, so that someone (maybe me) implements it in the future!

> Just take my view on this: Customizable colors can and do break the app. User choses wrong color, can't see anything anymore. Color swaps automatically from light to dark mode? Boom, broken. Requires two sets of colors. There are so many possibilities. Hence I prefer to keep it simple, have a sane default, whats here for years now. And not everybody can't be happy about that, but think the majority is OK with it.

I understand that more customization is not always for the better. Personally, I think in this situation though, it is warranted. FWIW, regarding users choosing the wrong colors, I would be far more concerned about the manual color selection of the editor FG/BG colors than about icon colors, but I get your point.

> These colors are used on multiple places. We don't make settings toggles for every single colorable item in the app.

I just thought of customizing the icon colors. If their color is reused somewhere else, ofc that should be kept in mind, but I think ideally there should be an option for just setting the icon colors (this is what this issue is about after all).

Also, I'm not here to try to make everything more colorful, I just want a more ergonomic UX. Even if there would be one toggle to color folder icons blue, even if I don't like the color, I would prefer it over the current state.

Since I don't know any Java it might take me a while to get to this, but I have it on my list.

**Comment by harshad1** [comment]
I think just updating our file and or folder icon to be more distinct solves 99% of this without any of the complexity of custom colors etc

**Comment by camoz** [comment]
Would be fine for me, but I guess existing users that are used to the current icons would want a way to restore the old looks, so I guess it would need to be a toggle ("color folder icons in blue" or something).

**Comment by camoz** [comment]
Here's a very simple mockup, in case the "one color solution" gets preferred over setting a custom color using a color picker. I just wanted to see what it could look like...

**Current default**

![markor-folderblue-idea_bw](https://github.com/gsantner/markor/assets/6276613/7013c08a-084b-4bc3-a613-5dd5aa253199)

**Blue folder icons**

![markor-folderblue-idea_color](https://github.com/gsantner/markor/assets/6276613/ec2cfbbb-b338-4f8a-bd3c-6d442e549c66)

**Blue folder icons (less saturation)**

![markor-folderblue-idea_color-less-saturation](https://github.com/gsantner/markor/assets/6276613/86836635-76b7-4d94-bc0f-d5c830d6dc85)

**Comment by glocalglocal** [comment]
I came to say I find using the same color for folders and files very confusing but I see @camoz got here first. I keep tapping on the wrong items. If different colors are not an option, at least some sort of thatching or other pattern could change the color perception. Different shapes alone are not sufficient IME.
